pkg/securitycontext/provider.go

```diff
@@ -37,10 +37,8 @@ type SimpleSecurityContextProvider struct{}
 // The security context provider can make changes to the Config with which
 // the container is created.
 func (p SimpleSecurityContextProvider) ModifyContainerConfig(pod *api.Pod, container *api.Container, config *docker.Config) {
-	if container.SecurityContext == nil {
-		return
-	}
-	if container.SecurityContext.RunAsUser != nil {
+	effectiveSC := determineEffectiveSecurityContext(pod, container)
```

Comment:
Still need a nil check here if PSC and CSC aren't set but this otherwise matches up with what we were discussing.